src/Components/Shared/src/BaseJSDataStream.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Components
+{
+    internal abstract class BaseJSDataStream : Stream
+    {
```

Comment:
TBH I don't think we get all that much from `BaseJSDataStream` versus inlining its code into the two subclasses. Might it be worth doing that just to avoid having another concept and name to keep track of?

Reply:
Yeah, this is literally implementing `Stream`. I wouldn't add a new thing to the type hierarchy at this point.